### PR TITLE
updated ReadMe file of examples folder

### LINKED ISSUE #13060: `README.md` in `examples/` doesn't reflect the contents of the directory
See [README.md](https://github.com/bokeh/bokeh/blob/a21c56740191bf21c560b832d0e9eb8a8b0046e4/examples/README.md). This wasn't updated after recent changes to the structure of examples' directory.

### PR DESCRIPTION
**Pull request by Med16-11**
- [x] issues: fixes #13060

### CODE REVIEW

**Comment by bryevdv** [comment]
@Med16-11 Thank you this is great! I only have one small ask: we generally try to keep markdown and docstring line lenghts to around 80-100 characters. Can you split up some of the single long lines in this PR? (Make sure there's no trailing spaces)

File: examples/README.md [hunk 1]
```diff
@@ -6,46 +6,59 @@ This directory contains many examples of different ways to use Bokeh. As Bokeh h
...
+### [`advanced`](advanced/)
 
-This directory contains examples of Bokeh Apps, which are simple and easy to create web applications for data visualization or exploration.
+This folder contains examples of more advanced or complex Bokeh applications and use cases, beyond the basic plotting and visualization examples that are included in the examples folder.
 
-### [`custom`](custom/)
+### [`basic`](basic/)
 
-The content of this directory demonstrate how to add custom extensions to Bokeh. Check project's documentation [page](https://docs.bokeh.org/en/latest/docs/user_guide/extensions.html)
+Each example in the basic folder provides a clear and concise demonstration of how to create a specific type of visualization using Bokeh. These examples are intended for new users who are just getting started with Bokeh and want to learn how to create basic visualizations using the library.
```
Comment (bryevdv) [inline examples/README.md]: I find these examples a useful reference even as an expert :D 
```suggestion
Each example in the basic folder provides a clear and concise demonstration of how to create a specific type of visualization using Bokeh. These examples are intended for users who want to learn how to create basic visualizations using the library.
```

File: examples/README.md [hunk 2]
```diff
@@ -6,46 +6,59 @@ This directory contains many examples of different ways to use Bokeh. As Bokeh h
...
 
-### [`embed`](embed/)
+### [`integration`](integration/)
```
Comment (bryevdv) [inline examples/README.md]: Actually the "integration" directory contains examples that are used as integration tests, they will probably be moved at some point. I think this entire section can just be omitted.

File: examples/README.md [hunk 3]
```diff
@@ -6,46 +6,59 @@ This directory contains many examples of different ways to use Bokeh. As Bokeh h
...
+
+This folder contains a collection of examples that demonstrate the usage and functionality of various Bokeh objects and properties, including Bokeh models, properties, and tools.
+
+### [`server`](server/)
+
+This folder contains a collection of examples that demonstrate how to create interactive web applications using Bokeh server
+
+### [`styling`](styling/)
+
+his folder contains a collection of examples that demonstrate various ways to style Bokeh plots using CSS styles and themes
```
Comment (bryevdv) [inline examples/README.md]: typo
```suggestion
This folder contains a collection of examples that demonstrate various ways to style Bokeh plots using CSS styles and themes.
```

File: examples/README.md [hunk 4]
```diff
@@ -6,46 +6,59 @@ This directory contains many examples of different ways to use Bokeh. As Bokeh h
...
 the user_guide](https://docs.bokeh.org/en/latest/docs/user_guide/concepts.html#interfaces)
 
-### [`webgl`](webgl/)
+### [`reference`](reference/)
+
+This folder contains a collection of examples that demonstrate the usage and functionality of various Bokeh objects and properties, including Bokeh models, properties, and tools.
+
+### [`server`](server/)
+
+This folder contains a collection of examples that demonstrate how to create interactive web applications using Bokeh server
```
Comment (bryevdv) [inline examples/README.md]: punctuation 
```suggestion
This folder contains a collection of examples that demonstrate how to create interactive web applications using Bokeh server.
```

**Comment by Med16-11** [comment]
Hey @bryevdv ! Thanks for the review, I have made the suggested changes.

**Comment by Med16-11** [comment]
Hey, @bryevdv do you want me to make any other changes?

**Comment by bryevdv** [comment]
Hi @Med16-11 this looks good, I did forget one ask the other day: can you add linebreaks to keep the line lenths close to 80-100 characters? With that change things will be good to merge. 

Also, since you have submitted several PRs, would you like to be added to @bokeh/dev ?

**Comment by Med16-11** [comment]
hey @bryevdv ! I have added the line breaks and sure you can add me in bokeh/dev. Thanks for asking. I would like to know more about bokeh since I have submitted my final Outreachy application for bokeh.  I am a full time university student and contribute in projects during my free time, hopefully I can make some valuable contributions.

**Comment by Med16-11** [comment]
hey @bryevdv I made the changes you suggested and submitted the updated pull request, but I noticed that it has not been reviewed yet. I understand that you are busy, but I wanted to inquire if there is anything I can do to help move this process along. Please let me know if there are any further changes or revisions I can make to the pull request, or if there is anything else I can do to help facilitate the review process. Thanks.

**Comment by Med16-11** [comment]
Hey @bryevdv !Does failing these tests has something to do with #12384 issue?

**Comment by bryevdv** [comment]
Hi @Med16-11 Most of the failures were sporadic and went green after a re-run. The documentation job is failing because an upstream package was released the other day with an incompatibility that we have not had chance to resolve yet.

**Comment by Med16-11** [comment]
Got it. Could you merge it now?

**Comment by bryevdv** [comment]
Yes, now that everything else is green

**Comment by Med16-11** [comment]
Thanks, @bryevdv ! Also, I would like to work on a few more issues. It doesn't have to be related to documentation as I have experience with python and data visualization too. So Are there any issues you want me to work on?

**Pull request merged**